beancount/plugins/close_tree_test.py

```diff
@@ -0,0 +1,192 @@
+import unittest
+
+from beancount.plugins import close_tree
+from beancount.parser import options
+from beancount import loader
+
+
+def s(e):
+    return sorted(e, key=lambda x: (x.date, getattr(x, 'account', 'XXX')))
+
+
+class TestCloseTree(unittest.TestCase):
+    def test_empty_entries(self):
+        new_entries, _ = close_tree.close_tree([], options.OPTIONS_DEFAULTS.copy())
+        self.assertEqual([], new_entries)
+
+    def test_basic(self):
+        entries, _, _ = loader.load_string("""
+            2014-01-01 open Assets:XBank
+            2014-01-01 open Assets:XBank:AAPL
+            2015-01-01 close Assets:XBank
+        """, dedent=True)
+
+        expected, _, _ = loader.load_string("""
+            2014-01-01 open Assets:XBank
+            2014-01-01 open Assets:XBank:AAPL
+            2015-01-01 close Assets:XBank
+            2015-01-01 close Assets:XBank:AAPL
+        """, dedent=True)
+
+        actual, _ = close_tree.close_tree(entries, {})
+        for a, e in zip(s(actual), s(expected)):
+            self.assertEqual(a.date, e.date)
+            self.assertEqual(a.account, e.account)
```

Comment:
Here and everywhere else, is there a reason why `self.assertEqual(actual, expected)` does not work? Also, why is there the need to sort the actual and expected entries, can't the expected entries be in the right order?

Reply:
1. Doesn't work because metadata is not the same
2. Sorting the expected entries as the actual would have been sorted makes it unintuitive IMO.
```
            2019-01-01 open Expenses:Non-Retirement:Auto:Fit:Insurance
            2019-01-01 open Expenses:Non-Retirement:Auto:Fit:Gas
            ....
            2021-06-17 open Expenses:Non-Retirement:Auto:Fit
            2021-06-17 close Expenses:Non-Retirement:Auto:Fit:Insurance
            2021-06-17 close Expenses:Non-Retirement:Auto:Fit:Gas
            2021-06-17 close Expenses:Non-Retirement:Auto:Fit
```
The above is neither alphabetical, nor consistent (ordering of opens and closes), nor does the open/close pair intuitively appear next to each other.